tests/render/test_template.py

```diff
@@ -0,0 +1,151 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from datetime import timezone, datetime, timedelta
+from unittest.mock import MagicMock
+
+import jinja2
+import pytest
+from jinja2 import TemplateNotFound, DictLoader
+from phabricatoremails import PACKAGE_DIRECTORY
+from phabricatoremails.render.events.common import Recipient, Reviewer, ReviewerStatus
+from phabricatoremails.render.events.phabricator import (
+    RevisionCommentPinged,
+    Revision,
+    ReplyContext,
+)
+from phabricatoremails.render.template import (
+    TemplateStore,
+    Template,
+    _jinja_html,
+    _jinja_text,
+)
+
+
+def test_templates_end_with_newline():
+    # The templates need to end with a newline so that the MIME sections are properly
+    # distanced. If they don't have sufficient newlines, then email clients won't
+    # realize that the HTML and text sections are separate.
+    template_dir = PACKAGE_DIRECTORY / "phabricatoremails" / "render" / "templates"
+    template_files = [f for f in template_dir.glob("*/*/*.jinja2")]
+
+    if not len(template_files):
+        pytest.fail("Didn't find any template files - did this test move?")
```

Comment:
Would moving this test cause this to fail? (seems like `template_dir` isn't related to where this test is.

Reply:
Good question: this test can move and it'll still work, but if the _templates_ move (e.g.: from `phabricatoremails/templates` to `phabricatoremails/render/templates`, this assertion will make sure that the test is still accurate.

Reply:
If the templates move, this test is at risk of silently failing.
So, this test _depends_ on the "found template files?" check.

The error message _is_ incorrect though, good catch. I'll fix that